### PR TITLE
chore(deps): pin and update dependencies

### PR DESCRIPTION
**Pull request by joe128**
The dependencies have not been updated for a while. I updated it and tested it with all of the example bot's use cases.  
I also tested whether the dependencies could be changed using renovate and it worked (see [master/renovate.json](https://github.com/joe128/signalbot/blob/master/renovate.json)).
I have also created a dev container, if you are interested I can create another PR (see [addDevContainer](https://github.com/joe128/signalbot/tree/chore/addDevContainer/.devcontainer)).

### CODE REVIEW

**Comment by filipre** [comment]
Hi! Thank you! Do these updates require us to jump from Python 3.9 to Python 3.12? In general, I'd only bump the version if we require a new feature from that version or we are closer to EOL.

**Comment by joe128** [comment]
No, it also works with older Python versions (Unit-tests runs with both 3.9.20 - the version of the project, and with 3.11.10 - the one with which the tests are supposed to run).
But in my opinion, it is always beneficial to use current versions and if you want to consider renovate-bot, the Python version would also be updated.
The missing point in my test with renovate would be updating the Python-version in the github-workflow-actions "setup-python", but that shouldn't be a problem.

**Comment by joe128** [comment]
I have now also adapted the workflow files so that the tasks should run under Python 3.12.7. 
But if you prefer to stay on python 3.11, I also created a PR for it (#74 )

File: pyproject.toml [hunk 1]
```diff
@@ -18,19 +18,19 @@ maintainers = ["René Filip"]
...
-version = "0.10.1"
+version = "0.11.0"
 
 [tool.poetry.dependencies]
-APScheduler = "^3.9.1"
-aiohttp = "^3.8.1"
-python = "^3.9"
-redis = "^4.1.4"
-websockets = "^10.2"
+APScheduler = "3.10.4"
```
[the remark this answers is not in the document]
Comment (joe128) [reply pyproject.toml:24]: I think Copier takes a middle path and uses ">=" whenever possible, which also allows newer major versions (unlike carret).  
There are now also newer versions (Pyhton (3.13), Websocket (14.1), Flake, ...)  
So, how would you like this version upgrade to be? Should the project be up to date but open to newer versions? Then i would suggest dependency-specifications for example for websocket ">= 14.1"

[the remark this answers is not in the document]
Comment (Era-Dorta) [reply pyproject.toml:24]: That makes sense. I agree with having ">=" and keeping the project up to date and open to newer versions.

File: pyproject.toml [hunk 2]
```diff
@@ -18,19 +18,19 @@ maintainers = ["René Filip"]
 name = "signalbot"
 readme = "README.md"
 repository = "https://github.com/filipre/signalbot"
-version = "0.10.1"
+version = "0.11.0"
```
[the remark this answers is not in the document]
Comment (joe128) [reply pyproject.toml]: changed

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (joe128) [reply pyproject.toml]: i have updated all dependencies. A local test was successfully.

**Review by Era-Dorta — APPROVED** [review]
Thanks a lot for the addressing the comments.

**Pull request merged**